guide/interactions/modals.md

```diff
@@ -0,0 +1,128 @@
+# Modals
+
+With modals, you can create pop-up forms that allow users to provide you, with formatted inputs, through submissions. We'll cover how to create, show, and receive modal forms using discord.js!
+
+::: tip
+This page is a follow-up to the [interactions (slash commands) page](/interactions/slash-commands.md). Please carefully read those first so that you can understand the methods used in this section.
+:::
+
+## Building and responding with modals
+
+Unlike message components, modals aren't strictly components themselves. They're a callback structure used to respond to interactions.
+
+::: tip
+You can have a maximum of five `ActionRowBuilder`s per modal builder, and one `TextInputBuilder` within an `ActionRowBuilder`. Currently, you cannot use `SelectMenuBuilder`s or `ButtonBuilder`s in modal action rows builders.
+:::
+
+To create a modal you construct a new `ModalBuilder`. You can then use the setters to add the title.
+
+```js {1,7-13}
+const { ModalBuilder } = require('discord.js');
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isChatInputCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const modal = new ModalBuilder()
+			.setCustomId('myModal')
+			.setTitle('My Modal');
+
+		// TODO: Add components to modal...
+
+		await interaction.showModal(modal);
+	}
+});
+```
+::: tip
+The custom id is a developer-defined string of up to 100 characters.
+:::
+
+As you can see, we construct the modal and assign it a custom id and a title. Once our modal is constructed, we send it as a response to the interaction via `#showModal`.
+
+We're still missing one step - adding inputs. Adding inputs is similar to adding components to messages.
+
+::: warning
+If you're using typescript you'll need to specify the type of components your action row holds. This can be done by specifying the
+generic parameter in `ActionRowBuilder`.
```

Comment:
```suggestion
If you're using typescript you'll need to specify the type of components your action row holds. This can be done by specifying the generic parameter in `ActionRowBuilder`.
```
obsolete line break